src/reportengine/tests/test_executor.py

```diff
@@ -76,10 +76,13 @@ def nsspec(x, beginning=()):
         self.graph.add_node(mcall, inputs={gcall, hcall})
 
 
-    def _test_ns(self):
+    def _test_ns(self, promise=False):
```

Comment:
Is there a particular reason for calling this `promise`?

Reply:
I think it looked like a js promise objet to me